### ISSUE TITLE
LogstachTcpSocketAppender and mandatory "destination" property vs logback behaviour

### ISSUE THREAD

**Issue by brenuart** [logstash/logstash-logback-encoder#134]
Suppose the following configuration fragment:
````
<appender name="REMOTE" class="net.logstash.logback.appender.LogstashTcpSocketAppender">
    <destination>${destinations}</destination>
    ...
</appender>
````

If the `destinations` property cannot be resolved, Logback's default (and non configurable) behaviour is to default it to `<prop.name>_IS_UNDEFINED`. The `TcpSocketAppender` requires at least one destination and stops in error if not the case. Unfortunately, in this scenario, from the appender's point of view, a value IS specified - so the precondition checks pass and no error is raised. The appender will eventually start but always fail to connect to a remote destination.

**Comment by brenuart** [comment]
As a temporary workaround, I changed my appender declaration in logback.xml as follows:
````
<appender name="REMOTE" class="net.logstash.logback.appender.LogstashTcpSocketAppender">
    <destination>${destinations:- }</destination>
    ...
</appender>
````
The `destinations` property defaults to a single space if not defined. The appender interprets the blank string as an empty value and complains because of no destinations specified.

This workaround works only for the `destination` property of course. Problem is still there for the other mandatory properties.

**Comment by philsttr** [comment]
Could add a validation check for string values that end with `_IS_UNDEFINED`

Any other suggestions?

**Comment by brenuart** [comment]
After some research, it seems there is no way to change Logback's behaviour. Looks like checking for a _IS_UNDEFINED suffix is the only solution I'm afraid :-(

**Comment by philsttr** [comment]
Added validation check.  Will be available when 4.6 is released.

**Comment by philsttr** [comment]
4.6 has been released and is on its way to maven central
